packages/orm/src/EntityManager.ts

```diff
@@ -672,7 +673,10 @@ export class EntityManager<C = {}> {
           await recalcAsyncDerivedFields(this, todos);
 
           if (!skipValidation) {
-            await validate(todos);
+            // Run simple rules first b/c it includes not-null/required rules, so that then when we run
+            // `validateReactiveRules` next, the lambdas won't see invalid entities.
```

Comment:
curious: Why not? Will they be stripped out of the `todos` array?

Reply:
If any simple rules fail, we'll blow up the `flush` with a `ValidationError`, so just won't get to the 2nd step of calling the hint-based rules/lambdas.